packages/typespec-ts/test/modularUnit/modelsGenerator.spec.ts

```diff
@@ -1320,4 +1316,389 @@ describe("inheritance & polymorphism", () => {
       `
     );
   });
+
+  describe("should generate models for header parameters", () => {
+    it("union variants with string literals being used in contentType headers", async () => {
```

Comment:
Do we have a case for response header with named union?

Reply:
updated.